File: Searching/JavaCodes/Leetcode367.java
class Solution {
   public static boolean isPerfectSquare(int num) {
        int start=1,end=num;
        
        while(start<end){
            int mid=start+(end-start)/2;
            if((long)mid*mid==num){
                return true;
            }
            if((long)mid*mid>num){
                end=mid-1;
            }
            else{
                start=mid+1;
            }
        }
        return (long)start*start==num;
    }
}

File: Searching/PythonCodes/Leetcode367.py
class Solution(object):
    def isPerfectSquare(self, num):
        start, end = 1, num

        while start <= end:
            mid = start + (end - start) // 2
            square = mid * mid
            
            if square == num:
                return True
            elif square > num:
                end = mid - 1
            else:
                start = mid + 1
        
        return False
